### DISCUSSION TITLE
Request for wiki infrastructure

### DISCUSSION THREAD

**Discussion by lukflug** [OSDEV-WIK1/wiki#22] [Ideas]
There are a couple of things that may be very useful in the future:
* Macros for citation needed and clarification needed.
* Warnings that are visible to the reader (i.e. not comments only visible to people viewing the source) about shortcomings of a page or section (ideally each type of issue should have a variant for each).
  * "this page or section cites insufficient or no sources"
  * "this page or section is a stub and needs expansion"
  * "this page or section may be biased towards a particular point of view"
  * "this page or section may contain factual inaccuracies"
* There are a lot of instances of cross-referencing non-existent pages. This must either be forbidden, or there must be a similar mechanism to "red links" in MediaWiki, where there is a visual difference between cross-references that exist and ones that don't exist to readers (e.g. generation of links to non-existent pages could be suppressed somehow).

**Comment by ArsenArsen** [comment]
I'm not entirely sure how possible doing redlinks with static content is; I'd rather have them fail.

The rest will require me digging into some Ruby, to create macros in AsciiDoctor, but should be overall pretty easy.

**Comment by lukflug** [comment]
Another thing might be an edit button, which would redirect the reader to GitHub.

**Reply by ArsenArsen** [reply]
I thought I added a "view source" button or such

**Reply by lukflug** [reply]
If so, I can't find it.

**Reply by ArsenArsen** [reply]
![image](https://user-images.githubusercontent.com/7805050/178580327-2ea16de7-744b-42e5-9850-e59d5f8541d8.png)

**Reply by lukflug** [reply]
Thanks!

**Reply by lukflug** [reply]
Maybe it might be a good idea to redirect directly to the equivalent URL on the main branch, instead of the commit that the site was generated on. In particular, it would allow people to click the edit button on GitHub, a thing that was requested on the OSDev Discord server.

**Reply by ArsenArsen** [reply]
Does the hash not permit that?

**Reply by ArsenArsen** [reply]
Apparently not, huh
